include/linker/common-ram.ld

```diff
@@ -45,6 +45,19 @@
 	} GROUP_DATA_LINK_IN(RAMABLE_REGION, ROMABLE_REGION)
 #endif
 
+#if defined(CONFIG_HAS_DYNAMIC_DEVICES)
+	SECTION_DATA_PROLOGUE(device_handles,,)
+	{
+	__device_handles_start = .;
+#ifdef LINKER_DEVICE_HANDLES_PASS1
+	KEEP(*(SORT(.__device_handles_pass1*)));
+#else
+	KEEP(*(SORT(.__device_handles_pass2*)));
+#endif /* LINKER_DEVICE_HANDLES_PASS1 */
+	__device_handles_end = .;
+	} GROUP_DATA_LINK_IN(RAMABLE_REGION, ROMABLE_REGION)
+#endif 	/* CONFIG_HAS_DYNAMIC_DEVICES */
```

Comment:
note: would it be possible to avoid duplication?

Reply:
I tried to split it in a different file, the problem is that the file is included in both linker script fragments, `commo-ram.ld` and `common-rom.ld`. Without know which fragment the file is being included the symbol `CONFIG_HAS_DYNAMIC_DEVICES` is not enough. e.g:
```
#if defined(CONFIG_HAS_DYNAMIC_DEVICES)
	SECTION_DATA_PROLOGUE(device_handles,,)
	{
	__device_handles_start = .;
#ifdef LINKER_DEVICE_HANDLES_PASS1
	KEEP(*(SORT(.__device_handles_pass1*)));
#else
	KEEP(*(SORT(.__device_handles_pass2*)));
#endif /* LINKER_DEVICE_HANDLES_PASS1 */
	__device_handles_end = .;
	} GROUP_DATA_LINK_IN(RAMABLE_REGION, ROMABLE_REGION)
#else /* No dynamic devices */
	SECTION_DATA_PROLOGUE(device_handles,,)
	{
	__device_handles_start = .;
#ifdef LINKER_DEVICE_HANDLES_PASS1
	KEEP(*(SORT(.__device_handles_pass1*)));
#else
	KEEP(*(SORT(.__device_handles_pass2*)));
#endif /* LINKER_DEVICE_HANDLES_PASS1 */
	__device_handles_end = .;
	} GROUP_ROM_LINK_IN(RAMABLE_REGION, ROMABLE_REGION)
#endif
```

This would be end up defining __device_handles_start in both ram and rom independently of `CONFIG_HAS_DYNAMIC_DEVICES` value.  The snippet above can be simplified, but does not solve the problem.

Reply:
My original proposal was to strip out the internals of the 
```
SECTION_DATA_PROLOGUE()
{
   /* Strip everything in here */
} > GROUP_LINK_IN()
```
into the common file, leaving the `#if defined(CONFIG_HAS_DYNAMIC_DEVICES)` where it is.
Just to keep the rather messy device handle pass logic in only one location.

Reply:
ok, that is possible.